autogen/oai/anthropic.py

```diff
@@ -64,14 +85,44 @@ def __init__(self, **kwargs: Any):
             api_key (str): The API key for the Anthropic API or set the `ANTHROPIC_API_KEY` environment variable.
         """
         self._api_key = kwargs.get("api_key", None)
+        self._aws_access_key = kwargs.get("aws_access_key", None)
+        self._aws_secret_key = kwargs.get("aws_secret_key", None)
+        self._aws_session_token = kwargs.get("aws_session_token", None)
+        self._aws_region = kwargs.get("aws_region", None)
 
         if not self._api_key:
             self._api_key = os.getenv("ANTHROPIC_API_KEY")
 
-        if self._api_key is None:
-            raise ValueError("API key is required to use the Anthropic API.")
+        if not self._aws_access_key:
+            self._aws_access_key = os.getenv("AWS_ACCESS_KEY")
+
+        if not self._aws_secret_key:
+            self._aws_secret_key = os.getenv("AWS_SECRET_KEY")
+
+        if not self._aws_session_token:
+            self._aws_session_token = os.getenv("AWS_SESSION_TOKEN")
+
+        if not self._aws_region:
+            self._aws_region = os.getenv("AWS_REGION")
+
+        if self._api_key is None and (
+            self._aws_access_key is None
+            or self._aws_secret_key is None
+            or self._aws_session_token is None
+            or self._aws_region is None
+        ):
+            raise ValueError("API key or AWS credentials are required to use the Anthropic API.")
```

Comment:
also, aws mentioned, some models are only enabled in few regions. can we have a check on that?
`**Claude 3.5 Sonnet is currently available in the US East (N. Virginia) Region`

cc @qingyun-wu @marklysze

Reply:
Anthropic bedrock throws an error incase of model unavailability and that will be caught in the exception block in this file.

Reply:
sure, thanks